### PR TITLE
Added frontend testing using Jest

### LINKED ISSUE #91: Add test cases for FE utils
1. Read guideline [here](https://gist.github.com/djeck1432/f722f5fe518c80b248ff609627da25e5)
2. Create `test` folder in [frontend](https://github.com/djeck1432/spotnet/tree/main/frontend) directory 
3. Add test cases (positive and negative) for next files:
- [contract.js](https://github.com/djeck1432/spotnet/blob/main/frontend/src/utils/contract.js)
-  [transaction.js](https://github.com/djeck1432/spotnet/blob/main/frontend/src/utils/transaction.js)
- [wallet.js](https://github.com/djeck1432/spotnet/blob/main/frontend/src/utils/wallet.js)
4. Update [README.md](https://github.com/djeck1432/spotnet/blob/main/frontend/README.md) how to run the test cases

### PR DESCRIPTION
**Pull request by Dprof-in-tech**
closes #91

Screenshot 2024-10-26 at 12 09 25

### CODE REVIEW

**Comment by whateverfw** [comment]
.

File: frontend/test/unit/transaction.test.js
Comment (whateverfw) [inline frontend/test/unit/transaction.test.js:1]: extract all the references of backend URL (http://0.0.0.0:8000) to a separate constant, which should be isolated into some config/constants file 
(this URL should be shared across all test files)

File: frontend/test/unit/contract.test.js
Comment (whateverfw) [inline frontend/test/unit/contract.test.js:1]: extract all the references of backend URL (http://127.0.0.1:8000) to a separate constant, which should be isolated into some config/constants file
(this URL should be shared across all test files)

File: frontend/test/unit/wallet.test.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        .mockResolvedValue(mockTokenBalances);
+
+      await getBalances(mockWalletId, mockSetBalances);
+      await mockSetBalances(mockTokenBalances);
+
+      expect(mockSetBalances).toHaveBeenCalledWith([
+        { name: "ETH", balance: "1.0000", icon: "ETH-icon" },
+        { name: "USDC", balance: "2.0000", icon: "USDC-icon" },
+        { name: "STRK", balance: "3.0000", icon: "STRK-icon" },
+      ]);
```
Comment (whateverfw) [inline frontend/test/unit/wallet.test.js:137]: you should use `mockTokenBalances` constant here

File: frontend/test/unit/contract.test.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+    process.env.REACT_APP_BACKEND_URL = mockBackendUrl;
+
+    getDeployContractData.mockReturnValue({
+      contractData: "mockContractData",
+    });
+  });
+
+  describe("deployContract", () => {
+    test("should successfully deploy contract", async () => {
```
Comment (whateverfw) [inline frontend/test/unit/contract.test.js:30]: stick to the single approach across all the test files and use either `it` or `test` keyword everywhere. I'd suggest/prefer `it`

File: frontend/test/unit/wallet.test.js [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+  getBalances,
+  logout,
+} from "../../src/utils/wallet";
+import {
+  ETH_ADDRESS,
+  STRK_ADDRESS,
+  USDC_ADDRESS,
+} from "../../src/utils/constants";
+
+// Mock SVG imports
```
Comment (whateverfw) [inline frontend/test/unit/wallet.test.js:14]: remove this comment

File: frontend/test/unit/wallet.test.js [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    it("should update balances state with token balances", async () => {
+      const mockSetBalances = jest.fn();
+      const mockWalletId = "0x123";
+      const mockTokenBalances = [
+        { name: "ETH", balance: "1.0000", icon: "ETH-icon" },
+        { name: "USDC", balance: "2.0000", icon: "USDC-icon" },
+        { name: "STRK", balance: "3.0000", icon: "STRK-icon" },
+      ];
+
+      // Mock the getTokenBalances function
```
Comment (whateverfw) [inline frontend/test/unit/wallet.test.js:125]: remove this comment

File: frontend/test/unit/transaction.test.js [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+    it("should handle transaction errors correctly", async () => {
+      const mockError = new Error("Transaction failed");
+      connect.mockResolvedValueOnce({
+        isConnected: true,
+        account: {
+          execute: jest.fn().mockRejectedValue(mockError),
+        },
+      });
+
+      console.error = jest.fn(); // Mock console.error
```
Comment (whateverfw) [inline frontend/test/unit/transaction.test.js:101]: remove this comment

**Review by whateverfw — CHANGES_REQUESTED** [review]

**Review by whateverfw — CHANGES_REQUESTED** [review]
`unit` folder should be renamed to `utils`

**Comment by Dprof-in-tech** [comment]
all done ser

**Comment by whateverfw** [comment]
@Dprof-in-tech Please take a look into other comments

File: frontend/test/unit/contract.test.js
Comment (Dprof-in-tech) [reply frontend/test/unit/contract.test.js:1]: fixed

File: frontend/test/unit/transaction.test.js
Comment (Dprof-in-tech) [reply frontend/test/unit/transaction.test.js:1]: fixed

File: frontend/test/unit/wallet.test.js (hunk 1 above)
Comment (Dprof-in-tech) [reply frontend/test/unit/wallet.test.js]: done ser

File: frontend/test/unit/wallet.test.js (hunk 3 above)
Comment (Dprof-in-tech) [reply frontend/test/unit/wallet.test.js]: done ser

File: frontend/test/unit/wallet.test.js (hunk 4 above)
Comment (Dprof-in-tech) [reply frontend/test/unit/wallet.test.js]: done ser

File: frontend/test/unit/transaction.test.js (hunk 5 above)
Comment (Dprof-in-tech) [reply frontend/test/unit/transaction.test.js]: done ser

**Comment by Dprof-in-tech** [comment]
> @Dprof-in-tech Please take a look into other comments above

i'm sorry i had seen the reviw in my mail, thought it was just one , ha ha

but its all fixed now

**Review by whateverfw — APPROVED** [review]

**Review by djeck1432 — APPROVED** [review]
Looks good to me
